Sources/Memberlist.php

```diff
@@ -327,13 +327,17 @@ function MLAll()
 		$limit = $second_offset - ($memberlist_cache['num_members'] - $_REQUEST['start']) - ($second_offset > $memberlist_cache['num_members'] ? $cache_step_size - ($memberlist_cache['num_members'] % $cache_step_size) : 0);
 	}
 
+	$custom_fields_qry = '';
+	if (!empty($context['custom_profile_fields']['join'][$col]))
+		$custom_fields_qry = $context['custom_profile_fields']['join'][$col];
+
 	// Select the members from the database.
 	$request = $smcFunc['db_query']('', '
 		SELECT mem.id_member
 		FROM {db_prefix}members AS mem' . ($_REQUEST['sort'] === 'is_online' ? '
 			LEFT JOIN {db_prefix}log_online AS lo ON (lo.id_member = mem.id_member)' : '') . ($_REQUEST['sort'] === 'id_group' ? '
 			LEFT JOIN {db_prefix}membergroups AS mg ON (mg.id_group = CASE WHEN mem.id_group = {int:regular_id_group} THEN mem.id_post_group ELSE mem.id_group END)' : '') . '
-			' . (!empty($context['custom_profile_fields']['join']) ? implode(' ', $context['custom_profile_fields']['join']) : '') . '
+			' . $custom_fields_qry . '
```

Comment:
What need to be imploded?
you only need to join the custom fields when you sort them.

You could keep the imploded when you decide only to support postgres,
because pg ignored the join part when not needed,
but mysql is stupid,
so it do the left join even it's not needed.